src/runtime_src/core/edge/user/shim.cpp

```diff
@@ -495,13 +503,131 @@ xclLoadXclBin(const xclBin *buffer)
   return ret;
 }
 
+#if defined(XRT_ENABLE_LIBDFX)
+static int
+libdfxLoadAxlf(std::shared_ptr<xrt_core::device> core_dev, const axlf *top, const axlf_section_header *overlay_header, int& fd, int flags, int &dtbo_id)
```

Comment:
This function is too long.  I'd suggest creating a libdfdx namespace and refactor within this namespace.
```
#if defined(...)
namespace libdfdx {

blah
factor1(...)
{
}

blan
factor2(...)
{
  ...
}

static int 
load_axlf(...)
{
   ... 
};

}
#endif // defined
```